src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,474 @@
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated, "remote identity is authenticated");
+                IdentityValidator.AssertHasName(serverIdentity, TestConfiguration.HostTarget);
+            }
+        }
+
+        [Fact, OuterLoop]
+        [PlatformSpecific(PlatformID.Linux)]
+        public void NegotiateStream_StreamToStream_KerberosAuthDefaultCredentials_Success()
+        {
+            if (!_isKrbAvailable)
+            {
+                return;
+            }
+
+            VirtualNetwork network = new VirtualNetwork();
+
+            using (var clientStream = new VirtualNetworkStream(network, isServer: false))
+            using (var serverStream = new VirtualNetworkStream(network, isServer: true))
+            using (var client = new NegotiateStream(clientStream))
+            using (var server = new UnixGssFakeNegotiateStream(serverStream))
+            {
+                Assert.False(client.IsAuthenticated, "client is not authenticated before AuthenticateAsClient call");
+
+                Task[] auth = new Task[2];
+                string user = string.Format("{0}@{1}", TestConfiguration.KerberosUser, TestConfiguration.Realm);
+                string target = string.Format("{0}@{1}", TestConfiguration.HostTarget, TestConfiguration.Realm);
+                // Seed the default Kerberos cache with the TGT
+                UnixGssFakeNegotiateStream.GetDefaultKerberosCredentials(user, _fixture.password);
+                auth[0] = client.AuthenticateAsClientAsync(CredentialCache.DefaultNetworkCredentials, target);
+                auth[1] = server.AuthenticateAsServerAsync();
+
+                bool finished = Task.WaitAll(auth, TestConfiguration.PassingTestTimeoutMilliseconds);
+                Assert.True(finished, "Handshake completed in the allotted time");
+
+                // Expected Client property values:
+                Assert.True(client.IsAuthenticated, "client is now authenticated");
+                Assert.Equal(TokenImpersonationLevel.Identification, client.ImpersonationLevel);
+                Assert.True(client.IsEncrypted, "client stream is encrypted");
+                Assert.True(client.IsMutuallyAuthenticated, "mutual authentication is true");
+                Assert.False(client.IsServer, "client is not server");
+                Assert.True(client.IsSigned, "client stream is signed");
+                Assert.False(client.LeaveInnerStreamOpen, "inner stream is open");
+
+                IIdentity serverIdentity = client.RemoteIdentity;
+                Assert.Equal("Kerberos", serverIdentity.AuthenticationType);
+                Assert.True(serverIdentity.IsAuthenticated,"server identity is authenticated");
+                IdentityValidator.AssertHasName(serverIdentity, target);
```

Comment:
The validation block of this test and the previous one look the same (aside from a variable name and an assert's message).  Can/should they be factored out to a common set of verification state?